src/main/role.ts

```diff
@@ -0,0 +1,64 @@
+import { RowBasecampMappingMissingError } from "./error/rowBasecampMappingMissingError";
+import { getRowBasecampMapping } from "./row";
+
+export enum RoleStatus {
+    OBSOLETE,
+    SURVIVING,
+    NEW
+}
+
+/**
+ * Gets the roleTodoIdMap object from the RowBasecampMapping object.
+ * Used for downstream processing
+ * 
+ * @param row a list of all the current roles associated with the row including the lead role. This may be identical to the original roles
+ * @returns a map that associates role titles with basecamp todo ids
+ */
+export function getRoleTodoIdMap(row: Row) {
+    const savedRowBasecampMapping: RowBasecampMapping | null = getRowBasecampMapping(row);
+    if(savedRowBasecampMapping == null) {
+        throw new RowBasecampMappingMissingError("The rowBasecampMapping object is null! Unable to proceed with updating the todo!");
+    }
+    return savedRowBasecampMapping.roleTodoIdMap
+}
```

Comment:
This function takes a Row as a parameter. Should it belong in the `row.ts` file?

Reply:
Moved getRoleTodoIdMap() to the row.ts file